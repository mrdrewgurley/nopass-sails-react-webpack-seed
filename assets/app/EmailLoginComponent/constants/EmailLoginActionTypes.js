export const ATTEMPT_HANDSHAKE = 'ATTEMPT_HANDSHAKE'
export const HANDSHAKE_FAILURE = 'HANDSHAKE_FAILURE'
export const HANDSHAKE_SUCCESS = 'HANDSHAKE_SUCCESS'
