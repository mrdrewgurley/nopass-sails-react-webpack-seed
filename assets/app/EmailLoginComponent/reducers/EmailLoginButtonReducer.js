import { handleActions } from 'redux-actions'
import * as types from '../constants/EmailLoginActionTypes'

const DEFAULT = {
  class: 'enabled',
  label: 'SEND LOGIN EMAIL',
  state: true
}

const actionHandlers = {
  [types.ATTEMPT_HANDSHAKE]: state => ({
    ...state,
    class: 'disabled',
    label: 'VERIFICATION PENDING',
    state: false
  }),
  [types.HANDSHAKE_FAILURE]: state => ({
    ...state,
    class: 'uk-form-danger',
    label: 'SEND LOGIN EMAIL',
    state: true
  }),
  [types.HANDSHAKE_SUCCESS]: state => ({
    ...state,
    class: 'uk-form-success',
    label: 'VERIFICATION SUCCESS',
    state: false
  })
}

export default handleActions(actionHandlers, DEFAULT)
