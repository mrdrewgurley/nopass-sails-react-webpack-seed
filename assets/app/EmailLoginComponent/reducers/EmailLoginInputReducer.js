import { handleActions } from 'redux-actions'
import * as types from '../constants/EmailLoginActionTypes'

const DEFAULT = {
  class: 'enabled',
  state: true
}

const actionHandlers = {
  [types.ATTEMPT_HANDSHAKE]: state => ({
    ...state,
    class: 'disabled',
    state: false
  }),
  [types.HANDSHAKE_FAILURE]: state => ({
    ...state,
    class: 'uk-form-danger',
    state: true
  }),
  [types.HANDSHAKE_SUCCESS]: state => ({
    ...state,
    class: 'uk-form-success',
    state: false
  })
}

export default handleActions(actionHandlers, DEFAULT)
