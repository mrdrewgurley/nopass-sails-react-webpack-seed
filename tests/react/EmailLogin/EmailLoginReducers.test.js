import EmailLoginButtonReducer from '../../../assets/app/EmailLoginComponent/reducers/EmailLoginButtonReducer'
import EmailLoginInputReducer from '../../../assets/app/EmailLoginComponent/reducers/EmailLoginInputReducer'
import * as Types from '../../../assets/app/EmailLoginComponent/constants/EmailLoginActionTypes'

/**
 * React.EmailLoginReducers
 * @desc Tests for the Email Login Reducers
 */
describe('Email Login Component Reducers', () => {
  describe('Email Login Button Reducer', () => {
    it('should return the initial state', () => {
      expect(EmailLoginButtonReducer(undefined, {})).toMatchSnapshot()
    })

    it('should handle attemptHandshake action', () => {
      expect(
        EmailLoginButtonReducer(undefined, {
          type: Types.ATTEMPT_HANDSHAKE
        }))
        .toMatchSnapshot()
    })

    it('should handle handshakeFailure action', () => {
      expect(
        EmailLoginButtonReducer(undefined, {
          type: Types.HANDSHAKE_FAILURE
        }))
        .toMatchSnapshot()
    })

    it('should handle handshakeSuccess action', () => {
      expect(
        EmailLoginButtonReducer(undefined, {
          type: Types.HANDSHAKE_SUCCESS
        }))
        .toMatchSnapshot()
    })
  })

  describe('Email Login Input Reducer', () => {
    it('should return the initial state', () => {
      expect(EmailLoginInputReducer(undefined, {})).toMatchSnapshot()
    })

    it('should handle attemptHandshake action', () => {
      expect(
        EmailLoginInputReducer(undefined, {
          type: Types.ATTEMPT_HANDSHAKE
        }))
        .toMatchSnapshot()
    })

    it('should handle handshakeFailure action', () => {
      expect(
        EmailLoginInputReducer(undefined, {
          type: Types.HANDSHAKE_FAILURE
        }))
        .toMatchSnapshot()
    })

    it('should handle handshakeSuccess action', () => {
      expect(
        EmailLoginInputReducer(undefined, {
          type: Types.HANDSHAKE_SUCCESS
        }))
        .toMatchSnapshot()
    })
  })
})
