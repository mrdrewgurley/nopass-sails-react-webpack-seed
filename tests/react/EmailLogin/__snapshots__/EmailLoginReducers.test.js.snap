// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Login Component Reducers Email Login Button Reducer should handle attemptHandshake action 1`] = `
Object {
  "class": "disabled",
  "label": "VERIFICATION PENDING",
  "state": false,
}
`;

exports[`Email Login Component Reducers Email Login Button Reducer should handle handshakeFailure action 1`] = `
Object {
  "class": "uk-form-danger",
  "label": "SEND LOGIN EMAIL",
  "state": true,
}
`;

exports[`Email Login Component Reducers Email Login Button Reducer should handle handshakeSuccess action 1`] = `
Object {
  "class": "uk-form-success",
  "label": "VERIFICATION SUCCESS",
  "state": false,
}
`;

exports[`Email Login Component Reducers Email Login Button Reducer should return the initial state 1`] = `
Object {
  "class": "enabled",
  "label": "SEND LOGIN EMAIL",
  "state": true,
}
`;

exports[`Email Login Component Reducers Email Login Input Reducer should handle attemptHandshake action 1`] = `
Object {
  "class": "disabled",
  "state": false,
}
`;

exports[`Email Login Component Reducers Email Login Input Reducer should handle handshakeFailure action 1`] = `
Object {
  "class": "uk-form-danger",
  "state": true,
}
`;

exports[`Email Login Component Reducers Email Login Input Reducer should handle handshakeSuccess action 1`] = `
Object {
  "class": "uk-form-success",
  "state": false,
}
`;

exports[`Email Login Component Reducers Email Login Input Reducer should return the initial state 1`] = `
Object {
  "class": "enabled",
  "state": true,
}
`;
